src/kernel/context.h

```diff
@@ -24,8 +25,11 @@ struct Context {
     //! Declare default constructor and destructor that are not inline, so code
     //! instantiating the kernel::Context struct doesn't need to #include class
     //! definitions for all the unique_ptr members.
-    Context();
+    static util::Result<std::unique_ptr<Context>> MakeContext();
```

Comment:
nit: as the above comment applies to all struct methods, how about generalizing the docstring?

<details>
<summary>git diff</summary>

```diff
diff --git a/src/kernel/context.h b/src/kernel/context.h
index 84fdcae800..9712ed6018 100644
--- a/src/kernel/context.h
+++ b/src/kernel/context.h
@@ -18,13 +18,13 @@ namespace kernel {
 //!
 //! State stored directly in this struct should be simple. More complex state
 //! should be stored to std::unique_ptr members pointing to opaque types.
+//!
+//! Methods should not be inline, so code instantiating the kernel::Context struct
+//! doesn't need to #include class definitions for all the unique_ptr members.
 struct Context {
     //! Interrupt object that can be used to stop long-running kernel operations.
     util::SignalInterrupt interrupt;
 
-    //! Declare default constructor and destructor that are not inline, so code
-    //! instantiating the kernel::Context struct doesn't need to #include class
-    //! definitions for all the unique_ptr members.
     static util::Result<std::unique_ptr<Context>> MakeContext();
     ~Context();
 
```
</details>